src/core/net/udp6.cpp

```diff
@@ -424,7 +424,14 @@ Message *Udp::NewMessage(uint16_t aReserved) { return NewMessage(aReserved, Mess
 
 Message *Udp::NewMessage(uint16_t aReserved, const Message::Settings &aSettings)
 {
-    return Get<Ip6>().NewMessage(sizeof(Header) + aReserved, aSettings);
+    Message *message = Get<Ip6>().NewMessage(sizeof(Header) + aReserved, aSettings);
+
+    if (message != nullptr)
+    {
+        message->SetDefaultHostOrigin();
+    }
```

Comment:
I am not sure we want to set origin as host from `Udp::NewMessage()` in general?

This method may be used by some module within the OT stack itself (say MLE or MeshCoP, etc) and in this case the origin is the stack and Thread Netif itself?

Reply:
When I left such packets' origin to `kOriginThreadNetif` instead of `kOriginHost*`, I saw nodes having trouble establishing links with each other but I don't know the exact reason. Any idea about the cause of this issue?

Reply:
My understanding is that MLE packets used to have `kOriginHost*` when being processed in `HandleDatagram`.

Reply:
Now I'm using the `kOriginHostTrusted` as the default when allocating a message. There seem not many cases where we want to use `kOriginThreadNetif` (mostly in mesh forwarder).